frontends/p4/toP4/toP4.cpp

```diff
@@ -254,7 +254,7 @@ bool ToP4::preorder(const IR::Type_Specialized* t) {
 
 bool ToP4::preorder(const IR::Argument* arg) {
     if (!arg->name.name.isNullOrEmpty()) {
-        builder.append(arg->name.toString());
+        builder.append((cstring)arg->name);
```

Comment:
explicit casts are always an indication something is not right.  This should work with an implicit conversion and so the cast could just be removed?